src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -104,7 +104,7 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
       "JOIN session_template st ON st.reference = v.session_template_reference " +
       "WHERE p.code = :prisonCode AND " +
       "v.session_template_reference = :sessionTemplateReference AND " +
-      "Date(v.visit_start) = :sessionDate AND " +
+      "(v.visit_start >= :sessionDate AND v.visit_start < CAST(:sessionDate AS DATE) + CAST('1 day' AS INTERVAL)) AND " +
```

Comment:
understand more after the call we had. Just a minor comment could we put can this statement in brackets - CAST(:sessionDate AS DATE) + CAST(‘1 day’ AS INTERVAL)

Reply:
Would be good to see what the query planner does with this query - might point us towards ways we can improve it further